### DISCUSSION TITLE
How exactly should I use $NVIM_APPNAME

### DISCUSSION THREAD

**Discussion by ghost** [glacambre/firenvim#1564] [Q&A]
First of all, thanks all contributors' hard work. I'm new to vim and neovim so I don't quite understand about the usage of $NVIM_APPNAME in the readme. 

> make sure "NVIM_APPNAME" is appropriately set when you run firenvim#install().

After searching about it I learn that $NVIM_APPNAME is an environment variable and I think `vim.env.NVIM_APPNAME='...'` should work, since I use lazy.nvim to install firenvim, I put the command in the config property after installation. code as below
```
require("lazy").setup({
  {
    'glacambre/firenvim',

    -- Lazy load firenvim
    -- Explanation: https://github.com/folke/lazy.nvim/discussions/463#discussioncomment-4819297
    lazy = not vim.g.started_by_firenvim,
    config = function()
	vim.env.NVIM_APPNAME = 'firenvim'
	vim.cmd([[colorscheme morning]])
    end,
    build = function()
        vim.fn["firenvim#install"](0)
    end
  }
})
```
after this, I open text area in browser and the colorscheme shows correctly and `:echo $NVIM_APPNAME` in firenvim command line also response correctly which is 'firenvim', the problem is that the textarea still use the 'init.lua' file in my .confg/nvim directory instead of the init file in .config/firenvim directory. so what should be the correct usage of $NVIM_APPNAME?

**Comment by glacambre** [comment]
Hi, thanks for opening this issue, I did not realize how unclear these instructions were.

Le 21 septembre 2023 07:41:41 GMT+02:00, phaye ***@***.***> a écrit :
> so what should be the correct usage of $NVIM_APPNAME?

Here are the steps you need to follow:
1) Create $XDG_CONFIG_HOME/$NVIM_APPNAME/init.lua . Assuming you want $NVIM_APPNAME to be "firenvim" and your $XDG_CONFIG_HOME is set to the standard location, this would be ~/.config/firenvim/init.lua.
2) Add the code necessary to load the firenvim plugin to that init.lua. The lazy.nvim commands you are using seem fine.
3) In your shell, run "NVIM_APPNAME=firenvim nvim". This should trigger the code in the init.lua you created and install the firenvim plugin.
4) Check that $XDG_DATA_HOME/firenvim/firenvim (~/.local/share/firenvim/firenvim by default) is a script and contains "export NVIM_APPNAME=firenvim". If it does not, try running ":call firenvim#install()" in the neovim instance you started by running "NVIM_APPNAME=firenvim nvim".

Hopefully this should be enough to get things working, please do not hesitate to ask more questions if you're still struggling !

**Comment by justinmk** [comment]
>  I'm new to vim and neovim so I don't quite understand about the usage of $NVIM_APPNAME in the readme.

Side-note: if you're new to vim, why bother with NVIM_APPNAME ?  It's an advanced feature.

**Reply by SeleneCosmia** [reply]
It's not really particularly difficult, as long as you know how to set an environmental variable in your preferred command-line shell then you can configure $NVIM_APPNAME.
